src/verifier.rs

```diff
@@ -1,11 +1,24 @@
-use bellman::groth16::{verify_proof, PreparedVerifyingKey, Proof};
+use bellman::groth16::{verify_proof, prepare_verifying_key, PreparedVerifyingKey, Parameters, Proof};
 use zcash_primitives::jubjub::{edwards, PrimeOrder, JubjubEngine};
 use ff::Field;
 use bellman::SynthesisError;
 use crate::AuthRoot;
 
-// TODO: lifetime?
+/// Verify a proof using the given CRS, VRF input and output, and
+/// authentication root.
 pub fn verify<E: JubjubEngine>(
+    proving_key: &Parameters<E>,
```

Comment:
This is a mistake, a `Parameters<E>` aka proving key could be *much* larger than a `VerifyingKey<E>`.  I'll merge this and fix it though.

Reply:
In practice we should prepare the `VerifyingKey` when we deserialize the SRS and not even support verifying with an unprepared key.